demonstrations_v2/tutorial_qksd_qsp_qualtran/demo.py

```diff
@@ -0,0 +1,401 @@
+# define states: 
+#
+# .. math:: \ket{\psi_k} = T_i(H)\ket{\psi_0},
+#
+# where :math:`T_i` is the i-th Chebyshev polynomial. Then we define a Krylov subspace of dimension
+# :math:`D` as 
+#
+# .. math:: \mathcal{K} = \text{span}(\{\ket{\psi_k}\}_{k=0}^{D-1}).
+#
+# With the Hamiltonian and Krylov space defined, we can use
+# the `Lanczos method <https://quantum-journal.org/papers/q-2023-05-23-1018/pdf/>`_ classically
+# or `QKSD <https://arxiv.org/pdf/2407.14431>`_ to find :math:`\tilde{H}`
+# and :math:`\tilde{S}`, then solve for the QKSD ground-state,
+# 
+# .. math:: |\Psi_0\rangle = \sum_k c^0_k | \psi_k \rangle = \sum_{i=0}^{D-1}c_iT_i(H)\ket{\psi_0},
+#
+# where :math:`c_i` are the cofficients of the :math:`i`-th Chebyshev polynomial. We choose the
+# Chebyshev basis here because QSP directly implements Chebyshev polynomials. Other types of
+# functions need to be converted into Chebyshev polynomials to implement via QSP.
+#
+# Using QSP to directly create the QKSD ground-state
+# --------------------------------------------------
+#
+# Only using QKSD to calculate one- and two-particle reduced density matrices
+# results in a quadratic scaling of the number of expectation
+# values that need to be measured with respect to the Krylov dimension, :math:`D` [#Oumarou].
+# Instead, reference [#Oumarou] shows that it is possible to reduce this to a constant scaling by preparing
+# the QKSD ground-state, :math:`|\Psi_0\rangle`, via QSP and measuring individual terms of the Hamiltonian.
+# [TODO: clarify QSP explanation below. Too many unclear variables and not obvious how qsp implements a polynomial+no block encodings mentioned]
+# The QSP circuit, :math:`U_{qsp}`, is defined as a series of alternating iterates,
+# :math:`W`, and rotations, :math:`S`, such that:
+# 
+# .. math:: U_\text{qsp} = S(\phi_0)\prod_k^{d-1}{W(a)S(\phi_k)}
+# 
+# Choosing the right rotation angles, :math:`\phi`, causes :math:`U_\text{qsp}` to implement a 
+# Chebyshev polynomial such that:
+#
+# .. math:: U_{qsp}|\psi_0\rangle|\psi_a\rangle = \sum_{i=0}^{D-1}c_iT_i(H)\ket{\psi_0} = \Psi_0\rangle|\psi_a'\rangle
+#
+# This will prepare the QKSD ground-state by implementing the corresponding Chebyshev polynomial of
+# the block-encoded Hamiltonian. For more details, see
+# `Function Fitting using Quantum Signal Processing <https://pennylane.ai/qml/demos/function_fitting_qsp>`_.
+# Let's see how to do this in PennyLane.
+#
+# Defining the QSP circuit
+# ~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# To define the QSP circuit, we first create a function to implement the rotations, :math:`S`. The
+# iterate, :math:`W`, will be implemented via the :class:`~pennylane.PrepSelPrep` class. We then
+# create a QSP template to alternate between :math:`W` and :math:`S`. Alternating between these operations
+# will prepare a Chebyshev polynomial of the input block-encoded matrix.
+
+def rotation_about_reflection_axis(angle, wires):
```

Comment:
```suggestion
def rotation_about_reflection_axis(angle, wires):
    """Rotation operation, S(phi_k)"""
```

Some really short docstring like this would help tie this function into the text/formalism